### DISCUSSION TITLE
How to handle holidays for different religions or ethnicities

### DISCUSSION THREAD

**Discussion by bojmaliev** [spatie/holidays#104] [Q&A]
Hello,

I'm coming from North Macedonia.

My country have different holidays for different people that are coming from different religions or ethnicities

It divided in multiple categories.

For example we have holidays for:
- All citizens 
- Only for orthodox people
- Only for catholic people
- Only for muslim people
- Only for Albanian people
- Only for Serbian people
- etc....

What do you think?

**Comment by inDeev** [comment]
Are those ethnic holidays included in law by authorities?

**Reply by bojmaliev** [reply]
Yes

**Comment by inDeev** [comment]
So it could be handled by some suffix, e.g. standard holidays for North Macedonia cold be mk, orthodox mk:orthodox, etc. Easiest way could be add separate countries like NorthMacedoniaOrthodox. But I'm not sure if it is ideal solution, maybe better could be to create some parser on country string and have it in one country model.

**Reply by bojmaliev** [reply]
Yeah, I like the second option more

**Comment by inDeev** [comment]
Well, can I prepare PR for North Macedonia where will be $ethnicType available in getAllHolidays() through some universal params and you then fill up the holidays?

**Reply by bojmaliev** [reply]
Yup, sure. Thank you.

But it's not just by ethnic group, but by nationality too.

Maybe we can use different word for the variable? 

Also just to mention, for me as christian orthadox - Macedonian I celebrate:

All country holidays + the one that are for Macedonian nationality + all orthodox

**Reply by inDeev** [reply]
it will be on you, I'm preparing it with 'type' param, but you could provide any param name.
``` $holidays = Holidays::for(country: 'mk?type=orhodox')->get(); ```
or
``` $holidays = Holidays::for(NorthMacedonia::make(['type' => 'orthodox']))->get(); ```

and then inside NorthMacedonia.php 
```
$params = $this->params ?? [];
$ethnicType = $params['type'] ?? 'standard';

if ($ethnicType === 'standard') {
...
```

**Comment by inDeev** [comment]
[PR-168](https://github.com/spatie/holidays/pull/168)

**Reply by inDeev** [reply]
Declined by @Nielsvanpach ,sorry, you must fin another approach

**Reply by bojmaliev** [reply]
Thank you for your try. I will try to find another approach

**Comment by kohenkatz** [comment]
Israel has a similar issue - some secular holidays and some religious holidays. While most of the religious holidays are Jewish, there are also some Christian (Catholic and Orthodox), Muslim, and Druze holidays recognized by the government, but each group is only supposed to take its own holidays (other than the common secular holidays). Within the Jewish holidays, some of the recognized holidays are only celebrated by certain groups, such as Ethiopian *Sigd* and North African (Maghrebi) *Mimouna*.

I want to submit a PR to add holidays for Israel, but I think we need a good way to handle all these options in order to make it worthwhile.
